core/unix/inc/TUnixSystem.h

```diff
@@ -31,11 +31,30 @@
 #include "TTimer.h"
 #endif
 
+#include <thread>
+#include <memory>
+
+int StackTraceExec(void *);
+
 typedef void (*SigHandler_t)(ESignals);
 
+struct TStackTraceHelper {
```

Comment:
Class not part of the public interface need to either be defined in an unnamed namepace in the source file or within the ROOT::Internal namespace or as protected/private subclass of the using class.